### PR TITLE
refactor(noUnusedImports): report empty named imports

### PR DESCRIPTION
**Pull request by Conaclos**
## Summary

Fix https://github.com/biomejs/biome/issues/3574 and https://github.com/biomejs/biome/issues/4240

## Test Plan

I added a test and updated the snapshots.

### CODE REVIEW

File: crates/biome_js_analyze/tests/specs/correctness/noUnusedImports/invalid-unused-react.jsx.snap [hunk 1]
```diff
@@ -19,7 +20,7 @@ import React, { useEffect } from "x"
 ```
 invalid-unused-react.jsx:1:8 lint/correctness/noUnusedImports  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━
 
-  ! This import is unused.
+  ! These imports are unused.
```
Comment (ematipico) [inline crates/biome_js_analyze/tests/specs/correctness/noUnusedImports/invalid-unused-react.jsx.snap:23]: The message *was* correct. Maybe we could use a more broad message, for example, "Unused import(s) detected."

File: crates/biome_js_analyze/tests/specs/correctness/noUnusedImports/invalid-import-namespace.ts.snap [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ---
 source: crates/biome_js_analyze/tests/spec_tests.rs
 expression: invalid-import-namespace.ts
+snapshot_kind: text
```
Comment (ematipico) [inline crates/biome_js_analyze/tests/specs/correctness/noUnusedImports/invalid-import-namespace.ts.snap:4]: Is this new? Which version of `cargo-insta` are you using?

**Review by ematipico — APPROVED** [review]
Neat!!

File: crates/biome_js_analyze/tests/specs/correctness/noUnusedImports/invalid-import-namespace.ts.snap (hunk 2 above)
Comment (Conaclos) [reply crates/biome_js_analyze/tests/specs/correctness/noUnusedImports/invalid-import-namespace.ts.snap:4]: The latest one: cargo-insta 1.41.1

File: crates/biome_js_analyze/tests/specs/correctness/noUnusedImports/invalid-unused-react.jsx.snap (hunk 1 above)
Comment (Conaclos) [reply crates/biome_js_analyze/tests/specs/correctness/noUnusedImports/invalid-unused-react.jsx.snap]: Fixed :)

**Pull request merged**